### DISCUSSION TITLE
Arduino Micro with 4 Encoders to MIDI

### DISCUSSION THREAD

**Discussion by Schmitti1306** [tttapa/Control-Surface#762] [Q&A]
Hello, im totally new in work with arduino.

So i need a sketch that works with 4 encoders 

i hope someone can help me. thanks

actually i have 1 Encoder running with following sketch:

#include <Control_Surface.h> // Include the Control Surface library
 
// Instantiate a MIDI over USB interface.
USBMIDI_Interface midi;
 
// Instantiate a CCRotaryEncoder object
CCRotaryEncoder enc {
  {2, 3},       // pins
  MCU::V_POT_1, // MIDI address (CC number + optional channel)
  1,            // optional multiplier if the control isn't fast enough
};
 
void setup() {
  // Select the correct relative MIDI CC mode.
  // Options:
  //   - TWOS_COMPLEMENT (default)
  //   - BINARY_OFFSET
  //   - SIGN_MAGNITUDE
  //   - NEXT_ADDRESS
  // Aliases:
  //   - REAPER_RELATIVE_1
  //   - REAPER_RELATIVE_2
  //   - REAPER_RELATIVE_3
  //   - TRACKTION_RELATIVE
  //   - MACKIE_CONTROL_RELATIVE
  //   - KORG_KONTROL_INC_DEC_1
  RelativeCCSender::setMode(relativeCCmode::MACKIE_CONTROL_RELATIVE);
  Control_Surface.begin(); // Initialize Control Surface
}
 
void loop() {
  Control_Surface.loop(); // Update the Control Surface
}

**Comment by tttapa** [comment]
You can create an array of four encoders. This is explained in [Getting Started: 4. Add MIDI Control Elements](https://tttapa.github.io/Control-Surface-doc/Doxygen/d5/d7d/md_pages_Getting-Started.html#autotoc_md283).

You could also compare the [Multiple-Control-Change-Potentiometers.ino](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/d8d/Multiple-Control-Change-Potentiometers_8ino-example.html) example to the [Control-Change-Potentiometer.ino](https://tttapa.github.io/Control-Surface-doc/Doxygen/de/de4/Control-Change-Potentiometer_8ino-example.html) example.
